kuryr_kubernetes/controller/managers/prometheus_exporter.py

```diff
@@ -0,0 +1,135 @@
+            '/metrics', methods=['GET'], view_func=self.metrics)
+        self.headers = {'Connection': 'close'}
+        self._os_net = clients.get_network_client()
+        self._project_id = config.CONF.neutron_defaults.project
+        self._create_metrics()
+
+    def metrics(self):
+        """Provides the registered metrics"""
+        self._record_quota_free_count_metric()
+        self._record_ports_quota_per_subnet_metric()
+
+        collected_metric = generate_latest(self.registry)
+        return Response(collected_metric, mimetype='text/plain')
+
+    def record_pod_creation_metric(self, duration):
+        """Records pod creation duration to the registry"""
+        self.pod_creation_latency.observe(duration)
+
+    @classmethod
+    def get_instance(cls):
+        if not ControllerPrometheusExporter.instance:
+            ControllerPrometheusExporter.instance = cls()
+        return ControllerPrometheusExporter.instance
+
+    def run(self):
+        address = '::'
+        try:
+            LOG.info('Starting Prometheus exporter')
+            self.application.run(address, CONF.controller_exporter.port)
+        except Exception:
+            LOG.exception('Failed to start Prometheus exporter')
+            raise
+
+    def _record_quota_free_count_metric(self):
+        """Records Network resources availability to the registry"""
+        quota = self._os_net.get_quota(quota=self._project_id, details=True)
+        for resource in RESOURCES:
+            resource_quota = getattr(quota, resource, None)
+            quota_limit = resource_quota['limit']
+            quota_used = resource_quota['used']
+            labels = {'resource': resource}
+            availability = quota_limit - quota_used
+            if availability >= _NO_QUOTA:
+                self.quota_free_count.labels(**labels).set(availability)
+
+    def _record_ports_quota_per_subnet_metric(self):
+        """Records the ports quota per subnet to the registry"""
+        subnets = self._os_net.subnets(project_id=self._project_id)
+        for subnet in subnets:
+            total_num_addresses = 0
+            ports_availability = 0
+            for allocation in subnet.allocation_pools:
+                total_num_addresses += netaddr.IPRange(
+                    netaddr.IPAddress(allocation['start']),
+                    netaddr.IPAddress(allocation['end'])).size
+                ports_count = len(list(self._os_net.ports(
+                    network_id=subnet.network_id,
+                    project_id=self._project_id)))
+            labels = {'subnet_id': subnet.id, 'subnet_name': subnet.name}
+            ports_availability = total_num_addresses-ports_count
+            self.port_quota_per_subnet.labels(**labels).set(ports_availability)
+
+    def _create_metrics(self):
+        """Creates a registry and records a new Gauge metric"""
+        self.registry = CollectorRegistry()
+        net_quota_metric_name = 'kuryr_quota_free_count'
```

Comment:
The names of vars are almost as long as content, just pass it directly into the class constructor. ;)